test/unit/api/api-llm.test.js

```diff
@@ -121,6 +121,49 @@ tap.test('Agent API LLM methods', (t) => {
     })
   })
 
+  t.test('withLlmCustomAttributes', (t) => {
+    const { api } = t.context
+    helper.runInTransaction(api.agent, (tx) => {
+      t.context.agent.tracer.getTransaction = () => {
+        return tx
+      }
+
+      t.doesNotThrow(() => {
```

Comment:
there are many different tests in one here. can you split them out please? also missing a test where you call `withLlmCustomAttributes` and not within a transaction